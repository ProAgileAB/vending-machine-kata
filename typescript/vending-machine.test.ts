import {describe, it, expect} from 'vitest'
import VendingMachine from "./vending-machine";
import VendingMachinePrinter from "./vending-machine-printer";

const coins = {"penny": 1,
    "nickel": 5,
    "dime": 10,
    "quarter": 25}

describe("Vending Machine", () => {
    it("foo", () => {
        let vendingMachine = new VendingMachine()
        let printer = new VendingMachinePrinter(vendingMachine)
        expect(printer.print()).toMatchSnapshot()
    })

    it("accept coins", () => {
        const machine = new VendingMachine();
        expect(machine.display === "INSERT COIN")

        machine.insert_coin(coins["nickel"])

        expect(machine.balance).toEqual(5)
        expect(machine.coins).toEqual([5])
        expect(machine.display).toEqual("5")
    })
})

