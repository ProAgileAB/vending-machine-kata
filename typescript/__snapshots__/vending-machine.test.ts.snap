// Vitest Snapshot v1

exports[`Vending Machine > foo 1`] = `
"VendingMachine
"
`;

exports[`Vending Machine > initial state 1`] = `
"VendingMachine
"
`;
